core/src/main/java/org/jruby/RubyEnumerator.java

```diff
@@ -473,8 +473,8 @@ private SizeFn enumSizeFn(final ThreadContext context) {
         final RubyEnumerator self = this;
         return new SizeFn() {
             @Override
-            public IRubyObject size(IRubyObject[] args) {
-                return self.size(context);
+            public IRubyObject size(ThreadContext context, IRubyObject self, IRubyObject[] args) {
+                return this.size(context, self, args);
```

Comment:
Yeah that's why I had to use `self1`. `recv` makes it much better i will go ahead with that then :)